src/HSE/All.hs

```diff
@@ -139,44 +157,80 @@ runCpp (Cpphs o) file x = dropLine <$> runCpphs o file x
         dropLine (line1 -> (a,b)) | "{-# LINE " `isPrefixOf` a = b
         dropLine x = x
 
-
 ---------------------------------------------------------------------
 -- PARSING
 
--- | A parse error from 'parseModuleEx'.
+-- | A parse error.
 data ParseError = ParseError
     {parseErrorLocation :: SrcLoc -- ^ Location of the error.
-    ,parseErrorMessage :: String -- ^ Message about the cause of the error.
+    ,parseErrorMessage :: String  -- ^ Message about the cause of the error.
     ,parseErrorContents :: String -- ^ Snippet of several lines (typically 5) including a @>@ character pointing at the faulty line.
+    ,parseErrorSDocs :: [SDoc]    -- ^ Parse error messages as reported by ghc-lib.
     }
 
+-- | Combined 'hs-src-ext' and 'ghc-lib-parser' parse trees.
+data ParsedModuleResults = ParsedModuleResults {
+    pm_hsext  :: (Module SrcSpanInfo, [Comment]) -- hs-src-ext result.
+  , pm_ghclib :: SrcLoc.Located (HsSyn.HsModule HsSyn.GhcPs) -- ghc-lib-parser result.
+}
+
+-- 'ghc-lib-parser' related constants.
+
+fakeSettings :: DynFlags.Settings
```

Comment:
I suggest creating a separate GHC.Util or similar tree - I think there will be a lot of GHC specific stuff that mirrors the HSE specific stuff.

Reply:
✔️